### PR TITLE
S3 Compatible Storage, Backup Database with Cron, Spanish Translation

### PR DESCRIPTION
**Pull request by jarscr**
* S3 Compatible Storage
* Backup Database with Cron
* Spanish Translation

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@
 .idea
```
Comment (jarscr) [inline .gitignore:3]: Remove .vscode folder

File: config/configuration.sample.php [hunk 2]
```diff
@@ -25,7 +25,8 @@ class config
   public $s3Bucket = ""; //Your S3 bucket
   public $s3Region = ""; //S3 region
   public $s3FolderName = ""; //Foldername within S3 (can be emtpy)
-            
+  public $s3EndPoint = ""; //S3 EndPoint S3 Compatible (https://sfo2.digitaloceanspaces.com)
+
```
Comment (jarscr) [inline config/configuration.sample.php:31]: Endpoint for use Compatible S3 Systems

File: public/backup.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
+<?php
```
Comment (jarscr) [inline public/backup.php:1]: Backup Database local or S3

File: public/download.php [hunk 4]
```diff
@@ -119,6 +119,7 @@ function getFileFromS3(){
     $s3Client = new Aws\S3\S3Client([
         'version'     => 'latest',
         'region'      => $config->s3Region,
+        'endpoint' => $config->s3EndPoint,
```
Comment (jarscr) [inline public/download.php:122]: Use S3 compatibl for download files

File: src/core/class.fileupload.php [hunk 5]
```diff
@@ -92,6 +92,7 @@ function __construct()
                 [
                 'version'     => 'latest',
                 'region'      => $this->config->s3Region,
+                'endpoint'    => $this->s3EndPoint,
```
Comment (jarscr) [inline src/core/class.fileupload.php:95]: Endpoint for S3 compatible

File: config/configuration.sample.php [hunk 6]
```diff
@@ -18,6 +18,8 @@ class config
 
   /* Fileupload */
   public $userFilePath= "userfiles/"; //Local relative path to store uploaded files (if not using S3)
+
+  public $dbBackupPath= "backupdb/"; //Local relative path to store backup files, need permission 0777
```
Comment (jarscr) [inline config/configuration.sample.php:22]: Folder for local Backup

File: public/backup.php [hunk 7]
```diff
@@ -16,19 +16,20 @@
 
 $config = new leantime\core\config();
 
-
 function runBackup($backupFile, $config){
 
-    $backupPath = $config->userFilePath.'/'.$backupFile;
+    
+    $backupPath = $config->dbBackupPath.$backupFile;
```
Comment (jarscr) [inline public/backup.php:22]: Use variable from config for local backup

File: public/backup.php [hunk 8]
```diff
@@ -16,19 +16,20 @@
...
 
 
     switch ($worked) {
         case 0:
             return array('type'=>'success','msg'=> 'The Database ' .$config->dbDatabase .' is save in the path '.getcwd().'/' .$backupPath );
+            chmod(ROOT.'/'.$config->userFilePath,0755);
             break;
         case 1:
-            return array('type'=>'error','msg'=>'There was an error backup ' .$config->dbDatabase . ' to ' . getcwd() . '/' . $backupPath);
+            return array('type'=>'error','msg'=>'There was an error backup ' .$config->dbDatabase . ' to ' . $backupPath);
```
Comment (jarscr) [inline public/backup.php:32]: Error control for local backup

File: public/backup.php [hunk 9]
```diff
@@ -56,7 +57,7 @@ function uploadS3($backupFile, $config){
         $result = $s3Client->putObject([
             'Bucket' => $config->s3Bucket,
             'Key'    => $config->s3FolderName . '/backupdb/' . $backupFile,
-            'Body'   => fopen($config->userFilePath.'/'.$backupFile, 'r'),
+            'Body'   => fopen($config->dbBackupPath.$backupFile, 'r'),
```
Comment (jarscr) [inline public/backup.php:60]: Use param from config for local backup

File: public/backup.php [hunk 10]
```diff
@@ -67,7 +68,7 @@ function uploadS3($backupFile, $config){
 
 }
 
-
+$S3=NULL;
```
Comment (jarscr) [inline public/backup.php:71]: If S3 is null don't remove local copy backup

File: public/backup.php [hunk 11]
```diff
@@ -78,7 +79,7 @@ function uploadS3($backupFile, $config){
 
     if($run['type']=="success"){
         $S3 = uploadS3($backupFile, $config);
-        @unlink($config->userFilePath.'/'.$backupFile);
+        @unlink($config->dbBackupPath.$backupFile);
```
Comment (jarscr) [inline public/backup.php:82]: Remove local backup if S3 is running

**Review by jarscr — COMMENTED** [review]
Local Backup options

File: resources/language/es-ES.ini [hunk 12]
```diff
@@ -12,6 +12,12 @@ language.currency="€"
 language.firstDayOfWeek="1"
 language.isMeridian="true"
 
+# special
```
Comment (jarscr) [inline resources/language/es-ES.ini:15]: Spanish Translate

**Review by jarscr — COMMENTED** [review]
Spanish Translation

File: public/backup.php [hunk 13]
```diff
@@ -84,10 +84,6 @@ function uploadS3($backupFile, $config){
 }else{
     $run = runBackup($backupFile, $config);
     $S3=NULL;
```
Comment (jarscr) [inline public/backup.php:87]: Fixed in the next commit

**Review by jarscr — COMMENTED** [review]
Please don't use this

File: public/backup.php [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+}else{
+    $run = runBackup($backupFile, $config);
+    $S3=NULL;
+}
+
+header('Content-Type: application/json');
+echo json_encode(array('backup' => $run['msg'], 's3' => $S3));
+
```
Comment (jarscr) [inline public/backup.php:92]: Use cron to excecute this and show json data

**Review by marcelfolaron — APPROVED** [review]
Thank you!

**Pull request merged**